src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2270,16 +2267,18 @@ bool Session::client_reset_if_needed()
                     m_progress.download.last_integrated_client_version);
     REALM_ASSERT_EX(m_progress.upload.client_version == 0, m_progress.upload.client_version);
 
-    m_upload_progress = m_progress.upload;
-    m_download_progress = m_progress.download;
+    // Reset the cached values which are used to calculate progress since the
+    // last time sync completed
     init_progress_handler();
-    // In recovery mode, there may be new changesets to upload and nothing left to download.
-    // In FLX DiscardLocal mode, there may be new commits due to subscription handling.
-    // For both, we want to allow uploads again without needing external changes to download first.
-    m_delay_uploads = false;
 
-    // Checks if there is a pending client reset
-    handle_pending_client_reset_acknowledgement();
+    // Update the download progress to match what it would have been if we'd
+    // received a MARK message from the server (as the fresh Realm which we used
+    // as the source data for the reset did).
+    m_upload_progress = m_progress.upload;
+    m_download_progress = m_progress.download;
+    m_server_version_at_last_download_mark = m_progress.download.server_version;
+    m_last_download_mark_received = m_last_download_mark_sent = m_target_download_mark;
+    check_for_download_completion();
```

Comment:
What are your thoughts on setting `m_reliable_download_progress` to `true` here so the call to `check_for_download_completion ()` also sends a progress notification report if there are no changes to be uploaded? Otherwise, we would have to wait for a download message to be received, which usually doesn't happen immediately after a client reset with no recovered local changes.

Reply:
Ignore this - the PBS behavior is incorrect and we will no longer send an empty progress notification at the end of a client reset with no pending local recovered changes.